mail_tracking/models/ir_mail_server.py

```diff
@@ -23,7 +23,7 @@ def _tracking_email_id_body_get(self, body):
         tracking_email_id = False
         # https://regex101.com/r/lW4cB1/2
         match = re.search(
-            r'<img [^>]* data-odoo-tracking-email=["\']([0-9]*)["\']', body)
+            r'<img[^>]*data-odoo-tracking-email=["\']([0-9]*)["\']', body)
```

Comment:
I find all of this would be much easier to maintain and understand by using `lxml.html` to parse the email body, and then XPath to find/remove the node. This regexp for instance would produce false positives with i.e. `<img src="i'm a hacker>"/>`, and given those come from the outside world, it's a good attack vector.

Reply:
Yes, this is another way to do it.
I think regex is faster and consume less memory. BTW, that vector is not matching the regex, see https://regex101.com/r/lW4cB1/3

Reply:
- `<img data-odoo-tracking-email="0'/>`

Reply:
Well, in those cases lxml will break, isn't it?

This fix first remove any tag that match with this regex, then inject the tag and after read the tag, so if there is some broken tag is not a security problem. 

Using lxml, what would the code have to do if no valid HTML is read?

Reply:
I think it should except and pass in such case, don't you think?

Reply:
I think that using lxml and then except and pass when invalid tags in the email will miss a tracking for some emails. In my approach (btw, the current one) those corner cases will be tracked without any problem.

So, I prefer to keep the current implementation because:
* It's more resilience
* It has a better CPU & RAM performance
* It's simpler
* It's the current implementation, less code changes
* It hasn't any security issue

Any other opinions here?